tests/js/server/aql/aql-graph-all-shortest-path.js

```diff
@@ -0,0 +1,450 @@
+        isPathValid(result[i], 4);
+      }
+    },
+
+    testNoPathExistsLimit: function () {
+      const query = `
+        FOR path IN OUTBOUND ALL_SHORTEST_PATHS "${source}" TO "${badTarget}" GRAPH "${graphName}"
+          LIMIT 3
+          RETURN path
+      `;
+      const result = db._query(query).toArray();
+      assertEqual(result.length, 0);
+    },
+
+    testFewerPathsThanLimit: function () {
+      const query = `
+        FOR path IN OUTBOUND ALL_SHORTEST_PATHS "${source}" TO "${target}" GRAPH "${graphName}"
+          LIMIT 1000
+          RETURN path
+      `;
+      const result = db._query(query).toArray();
+      allPathsDiffer(result);
+      assertEqual(result.length, 5);
+      for (let i = 0; i < 5; ++i) {
+        isPathValid(result[i], 4);
+      }
+    },
+
+    testPathsExistsNoLimit: function () {
+      const query = `
+        FOR source IN ${vName}
+          FILTER source._id == "${source}"
+          FOR target IN ${vName}
+            FILTER target._id == "${target}"
+            FOR path IN OUTBOUND ALL_SHORTEST_PATHS source TO target GRAPH "${graphName}"
+            RETURN path
+      `;
+      const result = db._query(query).toArray();
+      allPathsDiffer(result);
+      assertEqual(result.length, 5);
+      for (let i = 0; i < 5; ++i) {
+        isPathValid(result[i], 4);
+      }
+    },
+
+    testNoPathsExistsNoLimit: function () {
+      const query = `
+        FOR source IN ${vName}
+          FILTER source._id == "${source}"
+          FOR target IN ${vName}
+            FILTER target._id == "${badTarget}"
+            FOR path IN OUTBOUND ALL_SHORTEST_PATHS source TO target GRAPH "${graphName}"
+            RETURN path
+      `;
+      const result = db._query(query).toArray();
+      assertEqual(result.length, 0);
+    },
+
+    testPathsSkip: function () {
+      const query = `
+        FOR path IN OUTBOUND ALL_SHORTEST_PATHS "${source}" TO "${target}" GRAPH "${graphName}"
+          LIMIT 3, 3
+          RETURN path
+      `;
+      const result = db._query(query).toArray();
+      allPathsDiffer(result);
+      assertEqual(result.length, 2);
+
+      for (let i = 0; i < 2; ++i) {
+        isPathValid(result[i], 4);
+      }
+    },
+
+    testPathsSkipMoreThanExists: function () {
+      const query = `
+        FOR path IN OUTBOUND ALL_SHORTEST_PATHS "${source}" TO "${target}" GRAPH "${graphName}"
+          LIMIT 1000, 2
+          RETURN path
+      `;
+      const result = db._query(query).toArray();
+      assertEqual(result.length, 0);
+    },
+
+    testMultiDirections: function () {
+      const query = `
+        WITH ${vName}
+        FOR path IN OUTBOUND ALL_SHORTEST_PATHS "${source}" TO "${target}" ${e1Name}, INBOUND ${e2Name}
+          RETURN path
+      `;
+      const result = db._query(query).toArray();
+      allPathsDiffer(result);
+      assertEqual(result.length, 3);
+      for (let i = 0; i < 3; ++i) {
+        isPathValid(result[i], 3, true);
+      }
+    }
+
```

Comment:
I would like to request a quick and simple test here, where SOURCE and TARGET are Identical.
The underlying C++ mechanism can do it, but I think I spottet a potential glitch in the grammar (see grammar.y comment)
This may just work out of the box as the value is ignored, but better have a test here.

Reply:
done!

addressed in https://github.com/arangodb/arangodb/commit/7dd39c35d9422294200c784bbf804be15e0e7bab